screen/categories/src/main/java/com/ivy/categories/CategoriesViewModel.kt

```diff
@@ -171,9 +183,21 @@ class CategoriesViewModel @Inject constructor(
             }
 
             val sortedList = sortList(categories, sortOrder.value).toImmutableList()
+            this.allCategories.value = sortedList
+            this.categories.value = this.allCategories.value
+        }
+    }
 
-            this.categories.value = sortedList
+    private fun filterCategories(queryString: String) {
```

Comment:
No need for this, in pseudo code it should look like this

```kotlin
@Composable
fun getCategories() {
  val allCats = categories.value
  return remember(allCats, query.value) {
    allCats.filter { ... }.toImmutableList()
  }
}
```

Reply:
Done